geoportailv3/static/js/catalog/catalogdirective.js

```diff
@@ -45,21 +46,27 @@ app.module.directive('appCatalog', app.catalogDirective);
 
 /**
  * @constructor
+ * @param {angular.Scope} $scope Scope.
  * @param {app.Themes} appThemes Themes service.
  * @param {app.GetLayerForCatalogNode} appGetLayerForCatalogNode Function to
  *     create layers from catalog nodes.
  * @export
  * @ngInject
  */
-app.CatalogController = function(appThemes, appGetLayerForCatalogNode) {
+app.CatalogController = function($scope, appThemes,
+    appGetLayerForCatalogNode) {
 
-  appThemes.getThemeObject('main').then(goog.bind(
-      /**
-       * @param {Object} tree Tree object for the theme.
-       */
-      function(tree) {
-        this['tree'] = tree;
-      }, this));
+  $scope.$watch(goog.bind(function() {
+    return this['currentTheme'];
+  }, this), goog.bind(function(newVal, oldVal) {
```

Comment:
Very minor but you don't need the `newVal` and `oldVal` arguments. I know that, in certain cases, Angular applies optimizations when arguments are not specified.